core/views/MemberViews.py

```diff
@@ -25,12 +27,36 @@ def get_default_context(obj, context):
     return context
 
 
-class MemberDetailView(DetailView):
+class MemberListView(RestrictedViewList):
+
+    def can_view_all(self):
+
+        # A staffer can see all members
+        if self.request.user.is_staffer:
+            return True
+
+        # Non-staffers should only be able to see themselves
+        else:
+            self.restriction_filters["pk__exact"] = self.request.user.pk
+            return False
+
+
+class MemberDetailView(UserPassesTestMixin, DetailView):
     """Simple view that displays the all details of a user and provides access to specific change forms"""
 
     model = Member
     template_name = "admin/core/member/member_detail.html"
 
+    raise_exception = True
+    permission_denied_message = "You are not allowed to view another member's personal details!"
+
+    def test_func(self):
```

Comment:
What is a test doing here?

Reply:
It's not a test in the sense of unittests. Django calls test_func to check whether the member is allowed to access the view